esx_service/utils/auth_data.py

```diff
@@ -576,7 +638,9 @@ def connect(self):
 
         if not need_create_table:
             if os.path.isfile(self.db_path) and self.need_upgrade_db():
-                self.handle_upgrade_db()
+                error = self.handle_upgrade_db()
+                if error:
```

Comment:
@govint, for the first comment to change the upgrade check on service startup, I think it is out of the scope of this PR, we can file a issue for this and later to evaluate it whether we should make the change or not.
For the second part, if upgrade failed, you suggest that we should fallback to existing version. If will only work for the specific change we make since we can use the vm UUID if no vm name is available. But if later, we made some big schema change, then fallback to the existing version is not an option unless in the code we maintain different handling logic for old schema and new schema. 
CC @msterin